aspnet/mvc/views/forms/sample/final/Controllers/HomeController.cs

```diff
@@ -0,0 +1,120 @@
+﻿using FormsTagHelper.ViewModels;
+using Microsoft.AspNet.Mvc;
+
+namespace FormsTH
+{
+    public class HomeController : Controller
+    {
+        public IActionResult Index()
+        {
+            var model = new CountryViewModel();
+            model.Country = "CA";
+            return View(model);
+        }
+
+        [HttpPost]
+        [ValidateAntiForgeryToken]
+        public IActionResult Index(CountryViewModel model)
+        {
+            if (ModelState.IsValid)
+            {
+                string message = model.Country +  " selected";
+                return RedirectToAction("IndexSuccess", new {id=message});
+            }
+
+            // If we got this far, something failed, redisplay form.
```

Comment:
Good idea, I removed all but the first. BTW, this is a copy/paste/edit (I terminated the sentence with a period) from our template generated code. Probably not worth opening up a template bug.

// If we got this far, something failed, redisplay form